contracts/stellar-interchain-token-service/src/migrate.rs

```diff
@@ -0,0 +1,74 @@
+use soroban_sdk::Env;
+use stellar_axelar_std::interfaces::{CustomMigratableInterface, UpgradableClient};
+
+use crate::error::ContractError;
+use crate::storage::TokenIdConfigValue;
+use crate::types::CustomMigrationData;
+use crate::{storage, InterchainTokenService};
+
+pub mod legacy_storage {
+    use soroban_sdk::{contracttype, BytesN};
+    use stellar_axelar_std::contractstorage;
+
+    #[contractstorage]
+    enum LegacyDataKey {
+        #[temporary]
+        #[value(i128)]
+        FlowOut { flow_key: FlowKey },
+
+        #[temporary]
+        #[value(i128)]
+        FlowIn { flow_key: FlowKey },
+    }
+
+    #[contracttype]
+    #[derive(Clone, Debug)]
+    pub struct FlowKey {
+        pub token_id: BytesN<32>,
+        pub epoch: u64,
+    }
+}
+
+impl CustomMigratableInterface for InterchainTokenService {
+    type MigrationData = CustomMigrationData;
+    type Error = ContractError;
+
+    fn __migrate(env: &Env, migration_data: Self::MigrationData) -> Result<(), Self::Error> {
+        let CustomMigrationData {
+            new_token_manager_wasm_hash,
+            new_interchain_token_wasm_hash,
+            token_ids,
+            current_epoch,
+        } = migration_data;
+
+        for token_id in token_ids.into_iter() {
+            let TokenIdConfigValue {
+                token_address: interchain_token,
```

Comment:
This might be an existing token/asset, such as XLM. In this case, the interchain token contract can't/shouldn't be upgraded, only the token manager. Check the token manager type to determine if it should be upgraded.